compaction/compaction_manager.cc

```diff
@@ -956,6 +961,50 @@ void compaction_manager::submit(replica::table* t) {
     (void)perform_task(make_shared<regular_compaction_task>(*this, t));
 }
 
+bool compaction_manager::can_perform_regular_compaction(replica::table* t) {
+    return can_proceed(t) && !t->is_auto_compaction_disabled_by_user();
+}
+
+future<> compaction_manager::maybe_wait_for_sstable_count_reduction(replica::table* t) {
+    auto schema = t->schema();
+    if (!can_perform_regular_compaction(t)) {
+        cmlog.trace("maybe_wait_for_sstable_count_reduction in {}.{}: cannot perform regular compaction",
+                schema->ks_name(), schema->cf_name());
+        co_return;
+    }
+    auto num_runs_for_compaction = [&, this] {
+        auto& cs = t->get_compaction_strategy();
+        auto desc = cs.get_sstables_for_compaction(t->as_table_state(), get_strategy_control(), get_candidates(*t));
+        return boost::copy_range<std::unordered_set<utils::UUID>>(
+            desc.sstables
+            | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::run_identifier))).size();
```

Comment:
This is a weird way to count something. There got to be some `std::acumulate` or similar that is a better fit.

Reply:
I missed the the purpose of the set, indeed it doesn't seem like accumulate() would work here.